src/pal/src/arch/arm/exceptionhelper.S

```diff
@@ -22,12 +22,7 @@ LEAF_ENTRY ThrowExceptionFromContextInternal, _TEXT
     ldr	r10,	[r0, #(CONTEXT_R10)]
     ldr	r11,	[r0, #(CONTEXT_R11)]
     ldr	sp,	[r0, #(CONTEXT_Sp)]
-
-    ldr	r2,	[r0, #(CONTEXT_Pc)]
-
-    // Store return address to the stack
-    // Added r7 as a dummy to keep the stack aligned by 8 bytes.
-    push    {r2, r7}
+    ldr	lr,	[r0, #(CONTEXT_Pc)]
 
```

Comment:
Oh, thanks. I found the following comment from ``pal/src/exception/seh.cpp``:
```
    This function creates a stack frame right below the target frame, restores all callee
    saved registers from the passed in context, sets the RSP to that frame and sets the
    return address to the target frame's RIP.
    Then it uses the ThrowExceptionHelper to throw the passed in exception from that context.
```
It looks like that the semantics of this function requires one more stack frame to be established as you expalined.